### PR TITLE
Add typing definitions for bootstrap-menu

### PR DESCRIPTION
**Pull request by superheri**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by superheri** [comment]
If anyone can help me with this, I had a bit of problem with exporting the main function since it's not supposed to be what is exported by default and called directly when doing require...

File: types/bootstrap-menu/bootstrap-menu-tests.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import BootstrapMenu from "bootstrap-menu";
```
Comment (armanio123) [inline types/bootstrap-menu/bootstrap-menu-tests.ts]: This should be `import BootstrapMenu = require "bootstrap-menu"` instead.

File: types/bootstrap-menu/bootstrap-menu-tests.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+BootstrapMenu("#main", {
```
Comment (armanio123) [inline types/bootstrap-menu/bootstrap-menu-tests.ts]: The documentation shows this as a class instead. `var a = new BoostrapMenu(...)`

File: types/bootstrap-menu/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+export type BootstrapMenuEvent = "click" | "right-click" | "hover";
+
+export type BootstrapMenuRowElement = any;
+
+/**
+ * The function to mount the menu to an existing UI item
+ * @param selector Css selector of the element to attach the Menu to
+ * @param options Options to configure the menu
+ */
+export default function BootstrapMenu(selector: string, options: BootstrapMenuOptions): void;
```
Comment (armanio123) [inline types/bootstrap-menu/index.d.ts]: I believe this is incorrect as the package doesn't export a `default`. Instead you should use `export  =` syntax. `export = BootstrapMenu`.

Also is not a `function` but a class.

```ts
declare class BootstrapMenu {
    constructor(...)
}
```

Last, the rest of the exports can me added to a namespace:
```ts
declare namespace BoostrapMenu {
    export interface BootstrapMenuOptions {
        ...
    }
}
```

You can see more details on the [template documentation](https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-class-d-ts.html).

**Review by armanio123 — CHANGES_REQUESTED** [review]

File: types/bootstrap-menu/index.d.ts (hunk 3 above)
Comment (superheri) [reply types/bootstrap-menu/index.d.ts]: https://github.com/dgoguerra/bootstrap-menu/blob/master/src/BootstrapMenu.js#L431 For me this looks like a default export. If I'm not doing an export default I can't do directly new BootstrapMenu with the `import BootstrapMenu = require("bootstrap-menu")`

Comment (superheri) [reply types/bootstrap-menu/index.d.ts]: Ok my bad I just say that doing `export =` instead of `export class` works.

**Comment by superheri** [comment]
@armanio123 Is there a reason the build is failing even if I did like you told me?

**Comment by armanio123** [comment]
Sorry I forgot about tslint. If you check the travis logs, you'll notice this error: 
```
ERROR: 111:9  no-unnecessary-class  Every member of this class is a constructor. Use functions instead.
```
You an either avoid this disabling on `tslint.json` the rule, or if you want to go the extra mile, I noticed the  library also exposes a couple other functions on the class liek `init`, `updatePosition`, etc. You could add at least one method and this error will go away.

It's up to you. I'll approve any of the two solutions.

**Comment by superheri** [comment]
I added the `destroy` and `close` methods since it might be useful to call those to remove unused menus.

**Review by armanio123 — APPROVED** [review]

**Comment by armanio123** [comment]
Thanks for addressing the comments.

**Pull request merged**